protocols/virtualdefund/virtualdefund.go

```diff
@@ -199,22 +199,26 @@ func (o *Objective) initialOutcome() outcome.SingleAssetExit {
 	return o.V.PostFundState().Outcome[0]
 }
 
-func (o *Objective) generateFinalOutcome() outcome.SingleAssetExit {
+func (o *Objective) generateFinalOutcome() (outcome.SingleAssetExit, error) {
 	if o.MyRole != 0 {
-		panic("Only Alice should call generateFinalOutcome")
```

Comment:
Another example of something that "should never happen" unless there are bugs in the code.

Reply:
>Another example of something that "should never happen" unless there are bugs in the code.

I'm slightly on the fence about this once. You're right that it should never happen, but we only know it should never happen because the function is called in one place, guarded by an if statement.  If you're just reading `generateFinalOutcome` it's not clear why `o.MyRole != 0 ` will always be true.

Reply:
I guess when I say "it should never happen" I am not saying that it is impossible or even that I would bet money it wouldn't happen. What I mean more is, "if it does happen, that is fairly catastrophic and means we have a bug in our code". 

It's nice if we can _also_ become confident that it truly won't ever happen... and I take your point that in this case it does _not_ seem so safe to make that assumption.